hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -443,43 +429,36 @@ private void throwEntityNotFoundIfDsDoesNotExist(final Long distributionSetID) {
     }
 
     @Override
-    public long countByFilters(final Collection<TargetUpdateStatus> status, final Boolean overdueState,
-            final String searchText, final Long installedOrAssignedDistributionSetId,
-            final Boolean selectTargetWithNoTag, final String... tagNames) {
-        final List<Specification<JpaTarget>> specList = buildSpecificationList(new FilterParams(status, overdueState,
-                searchText, installedOrAssignedDistributionSetId, selectTargetWithNoTag, tagNames));
-        return countByCriteriaAPI(specList);
+    public long countByFilters(final FilterParams filter) {
+        List<Specification<JpaTarget>> specifications = buildSpecificationList(filter);
+        if (CollectionUtils.isEmpty(specifications)) {
```

Comment:
why use `CollectionUtils.isEmpty()` and not `specifications.isEmpty()`?

Reply:
historically grown ;-) - changed